File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cs
using System;
using System.Text;
using System.Collections.Generic;

public class Solution {
    public string solution(string s, string skip, int index) {
        var skipSet = new HashSet<char>(skip);
        var sb = new StringBuilder();
        
        int allowedCount = 26 - skipSet.Count;
        int step = index % allowedCount;
        
        foreach(char c in s)
        {
            char cur = c;
            int moved = 0;
            while(moved < step)
            {
                cur = (char)((((cur - 'a') + 1) % 26) + 'a');
                if(skipSet.Contains(cur)) continue;
                moved++;
            }
            sb.Append(cur);
        }
        return sb.ToString();
    }
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

string solution(string s, string skip, int index) {
    string answer = "";
    
    for(char &c : s)
    {
        int cnt = 0;
        while(cnt < index)
        {
            c = (c - 'a' + 1) % 26 + 'a';
            if(skip.find(c) == string::npos)
                cnt++;
        }
        answer += c;
    }
    return answer;
}